Sources/AsyncHTTPClient/ConnectionPool/HTTP2/HTTP2Connection.swift

```diff
@@ -241,7 +241,7 @@ final class HTTP2Connection {
                     // before.
                     let box = ChannelBox(channel)
                     self.openStreams.insert(box)
-                    self.channel.closeFuture.whenComplete { _ in
+                    channel.closeFuture.whenComplete { _ in
```

Comment:
I have run into this just today. If we send a too large header and the server sends a go away frame the child channel is closed but the parent channel not. This might be a good way to test this.

Reply:
I think we can try to test this by triggering the shutdown event and finding which channels get sent this user event. So long as we can hook the parent channel we should observe that it doesn't see it.

Reply:
The question is: How do I get access to the stream/child channel? All the ideas I had so far a quite locked down. Of course I can add an internal function that gives me the currently active channels.

```
func forTestingGetActiveChildChannels -> EventLoopFuture<[Channel]>
```

wdyt?

Reply:
I think adding a testing function is fine. You can guard it behind an SPI, which is probably sufficient.